include/api/CInferenceModelMetadata.h

```diff
@@ -35,12 +41,16 @@ class API_EXPORT CInferenceModelMetadata {
     using TVector = maths::CDenseVector<double>;
     using TStrVec = std::vector<std::string>;
     using TRapidJsonWriter = core::CRapidJsonConcurrentLineWriter;
+    using TPredictionFieldTypeResolverWriter =
+        std::function<void(const std::string&, EPredictionFieldType, TRapidJsonWriter&)>;
 
 public:
     //! Writes metadata using \p writer.
     void write(TRapidJsonWriter& writer) const;
     void columnNames(const TStrVec& columnNames);
     void classValues(const TStrVec& classValues);
+    void predictionFieldType(EPredictionFieldType predictionFieldType);
```

Comment:
I wouldn't bother to pass this in: where the callback is supplied the type is known so just wrap it into the callback.